### PR TITLE
Fix controller types

### PR DESCRIPTION
**Pull request by mmaruniak**

### CODE REVIEW

**Review by akincel — APPROVED** [review]

File: src/openApi/apiRequestModel.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export interface ApiRequest<T> {
-  body?: T;
+export interface ApiRequest<Body = any, Query = any> {
```
Comment (mmaruniak) [inline src/openApi/apiRequestModel.ts:1]: Soo initially the `T` was here to type the body, but then we needed to type the query params so (problably me) used T for either body or query. And this won't work for endpoints with both query & body and also having just `T` wasn't very clear that it's either body or query.

File: src/openApi/apiRequestModel.ts [hunk 2]
```diff
@@ -12,17 +12,21 @@ export interface ApiRequest<T> {
...
   headers: Record<string, string>;
   pathParameters: Record<string, string>;
-  queryStringParameters?: any;
+  queryStringParameters?: Query;
 }
 
-export interface PostRequest<T> extends ApiRequest<T> {
-  body: T;
+export interface PostRequest<Body = any, Query = any> extends ApiRequest<Body, Query> {
+  body: Body;
```
Comment (mmaruniak) [inline src/openApi/apiRequestModel.ts:19]: In theory, the body should still be optional as we have endpoints with an empty body. But then in the majority of controllers, we would have to null-check the body in each route. Instead, we can have it still required and pass `{}` if needed

File: src/openApi/openApiWrapper.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export default class OpenApiWrapper {
   constructor(requestLogger) {
     this.api = new OpenApi(
       {
-        requestMiddleware: (request: ApiRequest<unknown>): ApiRequest<unknown> => {
```
Comment (mmaruniak) [inline src/openApi/openApiWrapper.ts:31]: This is no longer `unknown` as we have to type our controllers now

File: src/openApi/apiRequestModel.ts (hunk 1 above)
Comment (akincel) [reply src/openApi/apiRequestModel.ts:1]: Yeah :)

File: src/openApi/apiRequestModel.ts (hunk 2 above)
Comment (akincel) [reply src/openApi/apiRequestModel.ts:19]: Who will need to pass `{}`? The HTTP clients of APIs?

Or is it "just" when writing unit tests or so?

Comment (mmaruniak) [reply src/openApi/apiRequestModel.ts:19]: Ah no just when you define the controller method. You can even so `PostRequest<null | undefined>` or even `PostRequest` if the body is not used. The `PostRequest<{}>` wasn't the best example

**Pull request merged**